lib/rate-limiter.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+const Bluebird = require('bluebird');
+
+exports.limit = (request, client, defaultRate) => {
+  if (!(request.route.method === 'get' || request.route.method === 'post' || request.route.method === 'delete')) {
+    return Bluebird.resolve(null);
+  }
+
+  let rate = defaultRate;
+  const routeSettings = request.route.settings.plugins ? request.route.settings.plugins.rateLimit : null;
+
+  if (routeSettings) {
+    rate = routeSettings.enabled !== false ? { limit: routeSettings.limit, window: routeSettings.window } : null;
+  }
+
+  if (rate) {
+    const key = `ratelimit:${request.auth.credentials.api_key}:${request.route.method}:${request.route.path}`;
```

Comment:
let's prefix with the name of the plugin `hapi-rate-limit` and make each section more and more granular:

`hapi-rate-limit:POST:/v1/postcards:live_12312124122312`